core/src/main/java/org/bitcoinj/wallet/WalletProtobufSerializer.java

```diff
@@ -446,10 +447,10 @@ public Wallet readWallet(InputStream input, boolean forceReset, @Nullable Wallet
         try {
             Protos.Wallet walletProto = parseToProto(input);
             final String paramsID = walletProto.getNetworkIdentifier();
-            NetworkParameters params = BitcoinNetworkParams.fromID(paramsID);
-            if (params == null)
+            Optional<BitcoinNetwork> network = BitcoinNetwork.fromIdString(paramsID);
+            if (!network.isPresent())
```

Comment:
This could be simplified with `.orElseThrow()`

Reply:
Oh yes. I'm always scared of picking a Java 9+ method when using these `else` methods.